### ISSUE TITLE
suggestion: prevent typo on some attributes

### ISSUE THREAD

**Issue by MoOx** [yannickcr/eslint-plugin-react#65]
I just lose 5min because I typed "propsTypes" instead of "propTypes". It might be cool to add some warning about that ?
I am sure there is other keywords we can handle.
Not sure if this should be handled by this plugin.

**Comment by bebraw** [comment]
Note that this problem changes form with ES6 type classes. In that case you would do something like

```javascript
export default class Note extends React.Component {
  constructor(props: {
    task: string;
    onEdit: Function;
  }) {
    super(props);

    this.state = {
      edited: false
    };
  }
}
```

Note that I've annotated my example using Flow but you get the idea. Overall ES6 definitions simplify things a lot since we can declare our defaults at the constructor.

**Comment by yannickcr** [comment]
I am afraid that typos handling produce a lot of false-positive errors. I find this kind of rule a bit hazardous and I'm not really comfortable with this :|

With the `prop-types` rule activated you should get a warning that the propTypes attribute is missing and quickly find out the problem.

**Comment by MoOx** [comment]
True say. I get some deactivated rules because of static properties in classes.
